r/R/record-batch-reader.R

```diff
@@ -255,6 +259,8 @@ as_record_batch_reader.arrow_dplyr_query <- function(x, ...) {
   # See query-engine.R for ExecPlan/Nodes
   plan <- ExecPlan$create()
   final_node <- plan$Build(x)
+  on.exit(plan$.unsafe_delete())
```

Comment:
Should we do the same everywhere else a plan is created? For example:

https://github.com/apache/arrow/blob/37a79659f82a7dbe0a55858e59fa5eaa380a266f/r/R/dataset-write.R#L153

https://github.com/apache/arrow/blob/37a79659f82a7dbe0a55858e59fa5eaa380a266f/r/R/query-engine.R#L381